code/concierge/src/main/kotlin/com/adobe/marketing/mobile/concierge/ui/stt/SpeechToTextPermissionHandler.kt

```diff
@@ -13,66 +13,148 @@
 import androidx.compose.runtime.setValue
+import androidx.compose.ui.platform.LocalContext
+import androidx.core.app.ActivityCompat
 
+/**
+ * Simple permission handler for audio recording.
+ * 
+ * Flow:
+ * 1. User clicks mic button (shouldRequestPermission = true)
+ * 2. Show rationale dialog explaining mic recording and Google speech recognition
+ * 3. Launch system permission dialog
+ * 4. Handle result:
+ *    - If allowed (once or while using) → Start recording
+ *    - If denied → Show settings dialog
+ */
 @Composable
 internal fun SpeechPermissionHandler(
     hasPermission: Boolean,
-    onPermissionResult: (Boolean) -> Unit
+    shouldRequestPermission: Boolean,
+    onPermissionResult: (Boolean) -> Unit,
+    onPermissionHandled: () -> Unit
 ) {
-    var showPermissionDialog by remember { mutableStateOf(false) }
+    var showRationaleDialog by remember { mutableStateOf(false) }
+    var showSettingsDialog by remember { mutableStateOf(false) }
+    var isInPermissionFlow by remember { mutableStateOf(false) }
+    val context = LocalContext.current
+    val activity = context as? Activity
 
+    // System permission launcher
     val permissionLauncher = rememberLauncherForActivityResult(
         contract = ActivityResultContracts.RequestPermission()
     ) { isGranted ->
-        onPermissionResult(isGranted)
+        if (isGranted) {
+            // Permission granted - notify parent
+            onPermissionResult(true)
+        } else {
+            // Permission denied - check if we can ask again
+            val canAskAgain = activity?.let {
+                ActivityCompat.shouldShowRequestPermissionRationale(
+                    it,
+                    Manifest.permission.RECORD_AUDIO)
+            } ?: false
+
+            if (canAskAgain) {
+                onPermissionResult(false)
+            } else {
+                // User denied with must go to settings next time
+                showSettingsDialog = true
+                onPermissionResult(false)
+            }
+        }
+        isInPermissionFlow = false
+        onPermissionHandled()
     }
 
-    if (showPermissionDialog) {
-        PermissionDialog(
-            onDismiss = { showPermissionDialog = false },
-            onConfirm = {
-                showPermissionDialog = false
-                permissionLauncher.launch(Manifest.permission.RECORD_AUDIO)
+    // Rationale dialog - shown before system permission dialog
+    if (showRationaleDialog) {
+        AlertDialog(
+            onDismissRequest = {
+                showRationaleDialog = false
+                isInPermissionFlow = false
+                onPermissionHandled()
+            },
+            title = { Text("Microphone & Speech Recognition Permission Required") },
+            text = { 
+                Text("This app needs access to your microphone to enable voice input. \n Your audio will be recorded and sent to Google servers for speech recognition processing.")
```

Comment:
there is a space added after the linebreak which is causing the text to be misaligned. we should remove the space to have the message text appear to be aligned with the first line of text.